### DISCUSSION TITLE
Add riscv64 builds to release workflow

### DISCUSSION THREAD

**Discussion by MichaIng** [navidrome/navidrome#2624] [Ideas]
As a step into the future and support making RISC-V more usable and popular, it would be great to have binaries released for this architecture. Buildx and Go toolchains should all support it, and someone seems to have succeeded compiling it: #1512

Probably it is not more needed than adding `linux/riscv64` to [pipeline.yml](https://github.com/navidrome/navidrome/blob/master/.github/workflows/pipeline.yml), [.goreleaser.yml](https://github.com/navidrome/navidrome/blob/master/.goreleaser.yml) and [pipeline.dockerfile](https://github.com/navidrome/navidrome/blob/master/.github/workflows/pipeline.dockerfile).

I could test builds on a VisionFive 2 and Star64 SBC. I can also try to test the builds and open a PR, when removing the DockerHub authentication and stuff (locally). In case just give me a hint whether there are other files with arch-specific content, which would require an addition.

**Comment by deluan** [comment]
The challenge is not Buildx or Go toolchains: It is adding the `taglib` binary and C++ cross-compilation toolkit to the image used to run gorelease: https://github.com/navidrome/ci-goreleaser/blob/master/Dockerfile

Unfortunately I don't have time to update this, but I'm open to accept a PR if anyone wants to tackle this.

**Reply by MichaIng** [reply]
Hmm, this cannot work with Debian Buster as base. To install `riscv64` packages, one need to use Debian Sid/unstable, e.g. this: https://hub.docker.com/layers/library/debian/sid-slim/images/sha256-8c9c82898528bb11740aca7e253e262bbfdf385493a95fdc54705931d522c4e1?context=explore

I could create a PR if hence a dedicated Dockerfile for those builds is okay.

**Reply by deluan** [reply]
If you get it to work, I'm open to review it and integrate in the pipeline, thanks!

**Comment by jack4942** [comment]
我在虚拟机Debian13通过riscv64-toolchains交叉编译成功Navidrome，并成功部署运行在VisionFive 2 和 Orange Pi RV2，详细可以看这篇https://blog.csdn.net/JackyLJ/article/details/157143570?spm=1001.2014.3001.5501。

**Comment by MichaIng** [comment]
I just opened a PR to implement `riscv64` support into the `navidrome/cross-taglib` CI: https://github.com/navidrome/cross-taglib/pull/17

And I prepared a commit to do the same for the Navidrome main repo, once above has been released: https://github.com/MichaIng/navidrome/pull/1

**Reply by deluan** [reply]
This is awesome! Thanks for the contribution. I'll review it and if all goes well, it will be part of next release (next week).

Keep in mind that a lot of this multi-platform code will go away, as we are moving to a "pure" Go version of taglib (actually based on WASM) in the next release. If all goes well, we will remove the CPP TagLib dependency in a future release

**Reply by deluan** [reply]
Ok, I released the new cross-taglib as 2.1.1-2, as the first part of the version must match the version of taglib used (2.1.1). Feel free to submit a PR for Navidrome main repo. Thanks!

**Reply by MichaIng** [reply]
Great, I updated the commit and opened a PR accordingly: #4949

**Reply by deluan** [reply]
Build successful. Are you able to test the binary? Also, can you test the docker image with alpine 3.20?
